charon/src/ast/values_utils.rs

```diff
@@ -155,12 +151,12 @@ impl ScalarValue {
     pub fn to_bits(&self) -> u128 {
         match *self {
             ScalarValue::Unsigned(_, v) => v,
-            ScalarValue::Signed(_, v) => u128::from_be_bytes(v.to_ne_bytes()),
+            ScalarValue::Signed(_, v) => u128::from_le_bytes(v.to_le_bytes()),
         }
     }
 
-    pub fn from_bytes(ty: IntegerTy, bytes: [u8; 16]) -> Self {
-        from_ne_bytes!(
+    pub fn from_le_bytes(ty: IntegerTy, bytes: [u8; 16]) -> Self {
+        from_le_bytes!(
```

Comment:
I assume we can't just use `u128::from_le_bytes`/`i128::from_le_bytes`; could you add a comment explaining why not if that's the case?

Reply:
I've added the following comment, that hopefully makes it clear:
```rust
/// Translates little endian bytes into a corresponding `ScalarValue`.
/// This needs to do the round-trip to the correct integer type to guarantee
/// that the values are correctly sign-extended (e.g. if the bytes encode -1i8, taking all 16 bytes
/// would lead to the value 255i128 instead of -1i128).
```